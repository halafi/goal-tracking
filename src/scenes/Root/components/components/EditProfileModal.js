// @flow

import * as React from 'react'
import { compose } from 'redux'
import { connect } from 'react-redux'
import { withFirebase } from 'react-redux-firebase'
import Button from 'material-ui/Button'
import TextField from 'material-ui/TextField'
import Dialog, { DialogActions, DialogContent, DialogTitle } from 'material-ui/Dialog'

import type { Profile } from '../../../../common/records/Firebase/Profile'

type Props = {
  firebase: any,
  profile: Profile,
  open: boolean,
  onClose: () => void,
}

type State = {
  name: string,
  url: string,
}

class EditProfileModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      name: props.profile.displayName,
      url: props.profile.photoURL || '',
    }
  }

  formValid = (): boolean => {
    const { name } = this.state

    // TODO validate, file upload
    return name.length >= 1
  }

  handleSubmit = (ev: any): void => {
    ev.preventDefault()

    const { firebase, onClose } = this.props
    const { name, url } = this.state

    if (!this.formValid()) {
      return
    }

    firebase.updateProfile({
      displayName: name,
      photoURL: url,
    })

    onClose()
  }

  handleClose = (): void => {
    const { onClose, profile } = this.props

    this.setState({
      name: profile.displayName,
      url: profile.photoURL || '',
    })

    onClose()
  }

  render() {
    const { open } = this.props
    const { name, url } = this.state

    return (
      <Dialog
        open={open}
        onClose={this.handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
        fullWidth
      >
        <DialogTitle id="alert-dialog-title">Edit Profile</DialogTitle>
        <DialogContent>
          <form onSubmit={this.handleSubmit} noValidate autoComplete="off">
            <TextField
              id="name"
              label="Display name"
              value={name}
              onChange={ev => this.setState({ name: ev.target.value })}
              margin="dense"
              fullWidth
            />
            <TextField
              id="photourl"
              label="Avatar Url"
              value={url}
              onChange={ev => this.setState({ url: ev.target.value })}
              margin="dense"
              fullWidth
            />
          </form>
        </DialogContent>
        <DialogActions>
          <Button
            disabled={!this.formValid()}
            onClick={this.handleSubmit}
            color="primary"
            autoFocus
          >
            Save
          </Button>
          <Button onClick={this.handleClose}>Cancel</Button>
        </DialogActions>
      </Dialog>
    )
  }
}

export default compose(
  withFirebase,
  connect(state => ({
    profile: state.firebase.profile,
  })),
  // withStyles(styles),
)(EditProfileModal)
