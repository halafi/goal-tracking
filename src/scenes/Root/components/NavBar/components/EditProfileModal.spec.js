import React from 'react'
import { shallow } from 'enzyme'

import EditProfileModal from './EditProfileModal'

describe('EditProfileModal', () => {
  it('renders component', () => {
    const wrapper = shallow(
      <EditProfileModal
        firebase={{ updateProfile: jest.fn() }}
        open
        onClose={jest.fn()}
        profile={{ userName: 'foo', displayName: 'bar' }}
      />,
    )
    expect(wrapper.getElement()).toMatchSnapshot()
    expect(wrapper.props()).toMatchSnapshot()
  })
})
