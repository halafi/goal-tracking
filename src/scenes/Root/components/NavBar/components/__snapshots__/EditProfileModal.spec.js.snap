// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileModal renders component 1`] = `
<WithStyles(Dialog)
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  fullWidth={true}
  onClose={[Function]}
  open={true}
>
  <WithStyles(DialogTitle)
    id="alert-dialog-title"
  >
    Edit Profile
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <form
      autoComplete="off"
      noValidate={true}
      onSubmit={[Function]}
    >
      <TextField
        fullWidth={true}
        id="name"
        label="Display name"
        margin="dense"
        onChange={[Function]}
        required={false}
        select={false}
        value="foo"
      />
      <TextField
        fullWidth={true}
        id="photourl"
        label="Avatar Url"
        margin="dense"
        onChange={[Function]}
        required={false}
        select={false}
        value=""
      />
    </form>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      disabled={false}
      onClick={[Function]}
    >
      Save
    </WithStyles(Button)>
    <WithStyles(Button)
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`EditProfileModal renders component 2`] = `
Object {
  "aria-describedby": "alert-dialog-description",
  "aria-labelledby": "alert-dialog-title",
  "children": Array [
    <WithStyles(DialogTitle)
      id="alert-dialog-title"
    >
      Edit Profile
    </WithStyles(DialogTitle)>,
    <WithStyles(DialogContent)>
      <form
        autoComplete="off"
        noValidate={true}
        onSubmit={[Function]}
      >
        <TextField
          fullWidth={true}
          id="name"
          label="Display name"
          margin="dense"
          onChange={[Function]}
          required={false}
          select={false}
          value="foo"
        />
        <TextField
          fullWidth={true}
          id="photourl"
          label="Avatar Url"
          margin="dense"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
        />
      </form>
    </WithStyles(DialogContent)>,
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        autoFocus={true}
        color="primary"
        disabled={false}
        onClick={[Function]}
      >
        Save
      </WithStyles(Button)>
      <WithStyles(Button)
        onClick={[Function]}
      >
        Cancel
      </WithStyles(Button)>
    </WithStyles(DialogActions)>,
  ],
  "fullWidth": true,
  "onClose": [Function],
  "open": true,
}
`;
